src/components/BackComponent/BackComponent.jsx

```diff
@@ -1,16 +1,35 @@
+import { useContext, useState } from "react";
 import { ThemeProvider } from "styled-components";
-import { useState } from "react";
-import WeatherCardComponent from "../WeatherCardComponent";
-import { BackgroundColor } from "./styled";
 import { rainyTheme, commonStyles } from "../../global/themes";
+import useCurrentWeather from "../../hooks/useCurrentWeather";
+import LocationContext from "../../context";
+import BackgroundColor from "./styled";
+import WeatherCardComponent from "../WeatherCardComponent";
+
+const initData = {
+  country: null,
+  city: null,
+  tempC: null,
+  condition: null,
+  humidity: null,
+  windKph: null,
+};
 
 const BackComponent = () => {
   const [theme] = useState({ ...rainyTheme, ...commonStyles });
+  const [weatherData, setWeatherData] = useState({
```

Comment:
i believe it would be better if you replace it in useCurrentWeather

Reply:
const {  weatherData, status, error  } = useCurrentWeather(latitude, longitude);